### PR TITLE
Add check-manifest to CI and release workflow

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
In playing around with some install tooling, I learned about [check-manifest](https://github.com/mgedmin/check-manifest), which is a nice tool that seems to be pretty respected.  Using it, I realized that we're not including a number of `_tests` folders because they don't include `__init__.py` (making our pre-release tests a little less covered).  This fixes that and adds `check-manifest` to our release pipeline so it will be hard to mistakenly miss a file again.

Note, I add the tests by adding `recursive-include napari _tests/*.py` to `MANIFEST.in`, which I _think_ means we could remove all of the `__init__.py` files in our `_tests` folders, but I have stopped short of that here.  @jni?
(edit: well, we have some relative imports in there ... so we can't remove _all_ of them)

## Type of change

- [x] packaging / support

# References
https://github.com/mgedmin/check-manifest
https://hynek.me/articles/sharing-your-labor-of-love-pypi-quick-and-dirty/

# How has this been tested?

- [x] all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sofroniewn** [comment]
Seems like something is going wrong with `check_requirements`

```
check-manifest pip-missing-reqs napari --ignore-file="**/_tests/*" --ignore-module=PySide2 --ignore-module=zarr  --ignore-file=napari/conftest.py
usage: check-manifest [-h] [--version] [-q] [-v] [-c] [-u] [-p PYTHON]
                      [--ignore patterns] [--ignore-bad-ideas patterns]
                      [source_tree]
check-manifest: error: unrecognized arguments: napari --ignore-file=**/_tests/* --ignore-module=PySide2 --ignore-module=zarr --ignore-file=napari/conftest.py

Exit status: 2
```

File: MANIFEST.in [hunk 1]
```diff
@@ -1,7 +1,17 @@
 include LICENSE
 include versioneer.py
 include napari/_version.py
+recursive-include napari _tests/*.py
 recursive-include napari/resources *.png *.jpeg *.jpg *.svg *.qss
 exclude napari/resources/_qt_resources*.py
 include napari/utils/colormaps/matplotlib_cmaps.txt
+
+exclude requirements/*.txt requirements.txt
```
Comment (jni) [inline MANIFEST.in:9]: Wait shouldn't requirements be included in our sdist?

Comment (jni) [reply MANIFEST.in:9]: Oh just saw the below. Ignore me! :sweat_smile:

File: MANIFEST.in [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
 include requirements/default.txt
+
+# explicit excludes to keep check-manifest happy and remind us that
+# these things are not being included unless we ask
+recursive-exclude docs *
+recursive-exclude examples *
+exclude napari/benchmarks/*
+recursive-exclude resources *
```
Comment (jni) [inline MANIFEST.in:17]: Cool. =)

File: pyproject.toml [hunk 3]
```diff
@@ -21,3 +21,12 @@ exclude = '''
...
 '''
+
+[tool.check-manifest]
+ignore = [
+  ".cirrus.yml",
+  ".pre-commit-config.yaml",
+  "asv.conf.json",
+  "codecov.yml",
+  "Makefile",
+  ]
```
Comment (jni) [inline pyproject.toml]:
```suggestion
]
```
?

File: .cirrus.yml [hunk 4]
```diff
@@ -227,6 +227,7 @@ codecov_task:
     pip install pytest-cov codecov
 
   check_requirements_script:
+    check-manifest
```
Comment (jni) [inline .cirrus.yml]: This should fix the failure, based on the above syntax:

```suggestion
  check_requirements_script: |
    check-manifest
```

**Review by jni — APPROVED** [review]
@tlambert03 Just one minor comment that should fix the CI, and a totally nitpicky and possibly incorrect formatting suggestion. Otherwise this looks great to me. :blush:

**Comment by jni** [comment]
I should mention that we've had a whole bunch of manifest snafus with skimage over the years, so I think this is a honking great idea!

File: .cirrus.yml (hunk 4 above)
Comment (tlambert03) [reply .cirrus.yml]: thanks that fixed it

**Review by sofroniewn — APPROVED** [review]
Love this, as someone who has messed this up in napari before, nice to have this in!!

**Pull request merged**